ssh.go

```diff
@@ -43,16 +44,23 @@ func (e ErrConnect) Error() string {
 
 // parseHost parses and normalizes <user>@<host:port> from a given string.
 func (c *SSHClient) parseHost(host string) error {
-	c.host = host
+	if !strings.Contains(host, "://") {
+		host = "ssh://" + host
+	}
+
+	info, err := url.Parse(host)
+	if err != nil {
+		return err
+	}
 
-	// Remove extra "ssh://" schema
-	if len(c.host) > 6 && c.host[:6] == "ssh://" {
-		c.host = c.host[6:]
+	c.host = info.Host
+	if u := info.User.Username(); u != "" {
+		c.user = u
 	}
 
-	if at := strings.Index(c.host, "@"); at != -1 {
-		c.user = c.host[:at]
-		c.host = c.host[at+1:]
+	// Add default port, if not set
+	if _, p, err := net.SplitHostPort(info.Host); err != nil && p == "" {
```

Comment:
this condition looks suspicious

should this be `err == nil && p == ""`?

or `err != nil || p == ""` ?

Reply:
oh, I's my mistake! Thanks for your review! :-)